[MSYS2] Unable to build Zephyr

Hi,

following the installation instructions I was unable to build Zephyr using MSYS2.

The error I get is:

> sebo@SEBO-WIN MINGW64 ~/zephyr/samples/hello_world/build
> $ cmake -GNinja -DBOARD=nrf52_pca10040 ..
> CMake Deprecation Warning at C:/msys64/home/sebo/zephyr/cmake/app/boilerplate.cmake:37 (cmake_policy):
>   The OLD behavior for policy CMP0000 will be removed from a future version
>   of CMake.
> 
>   The cmake-policies(7) manual explains that the OLD behaviors of all
>   policies are deprecated and that a policy should be set to OLD only under
>   specific short-term circumstances.  Projects should be ported to the NEW
>   behavior and not rely on setting a policy to OLD.
> Call Stack (most recent call first):
>   CMakeLists.txt:2 (include)
> 
> 
> ############### C:/msys64/home/sebo/zephyr
> ############### C:/msys64/home/sebo/zephyr
> -- Selected BOARD nrf52_pca10040
> C:/msys64/home/sebo/zephyr/VERSION
> CMake Error at C:/msys64/home/sebo/zephyr/cmake/version.cmake:5 (file):
>   file failed to open for reading (No such file or directory):
> 
>     /home/sebo/zephyr/samples/hello_world/C:/msys64/home/sebo/zephyr/VERSION
> Call Stack (most recent call first):
>   C:/msys64/home/sebo/zephyr/cmake/app/boilerplate.cmake:220 (include)
>   CMakeLists.txt:2 (include)
> 
> 
> C:/msys64/home/sebo/zephyr/VERSION
> CMake Error at C:/msys64/home/sebo/zephyr/cmake/version.cmake:8 (string):
>   string sub-command REGEX, mode MATCH needs at least 5 arguments total to
>   command.
> Call Stack (most recent call first):
>   C:/msys64/home/sebo/zephyr/cmake/app/boilerplate.cmake:220 (include)
>   CMakeLists.txt:2 (include)
> 
> 
> CMake Error at C:/msys64/home/sebo/zephyr/cmake/version.cmake:11 (string):
>   string sub-command REGEX, mode MATCH needs at least 5 arguments total to
>   command.
> Call Stack (most recent call first):
>   C:/msys64/home/sebo/zephyr/cmake/app/boilerplate.cmake:220 (include)
>   CMakeLists.txt:2 (include)
> 
> 
> CMake Error at C:/msys64/home/sebo/zephyr/cmake/version.cmake:14 (string):
>   string sub-command REGEX, mode MATCH needs at least 5 arguments total to
>   command.
> Call Stack (most recent call first):
>   C:/msys64/home/sebo/zephyr/cmake/app/boilerplate.cmake:220 (include)
>   CMakeLists.txt:2 (include)
> 
> 
> CMake Error at C:/msys64/home/sebo/zephyr/cmake/version.cmake:17 (string):
>   string sub-command REGEX, mode MATCH needs at least 5 arguments total to
>   command.
> Call Stack (most recent call first):
>   C:/msys64/home/sebo/zephyr/cmake/app/boilerplate.cmake:220 (include)
>   CMakeLists.txt:2 (include)
> 
> 
> CMake Error at C:/msys64/home/sebo/zephyr/cmake/version.cmake:20 (string):
>   string sub-command REGEX, mode MATCH needs at least 5 arguments total to
>   command.
> Call Stack (most recent call first):
>   C:/msys64/home/sebo/zephyr/cmake/app/boilerplate.cmake:220 (include)
>   CMakeLists.txt:2 (include)
> 
> 
> Zephyr version: ..
> CMake Error at C:/msys64/home/sebo/zephyr/cmake/version.cmake:35 (math):
>   math cannot parse the expression: "( << 16) + ( << 8) + ()": syntax error,
>   unexpected exp_SHIFTLEFT, expecting exp_PLUS or exp_MINUS or exp_OPENPARENT
>   or exp_NUMBER (4)
> Call Stack (most recent call first):
>   C:/msys64/home/sebo/zephyr/cmake/app/boilerplate.cmake:220 (include)
>   CMakeLists.txt:2 (include)
> 
> 
> CMake Error at C:/msys64/home/sebo/zephyr/cmake/version.cmake:36 (math):
>   math cannot parse the expression: "( << 16) + ( << 8) + ()": syntax error,
>   unexpected exp_SHIFTLEFT, expecting exp_PLUS or exp_MINUS or exp_OPENPARENT
>   or exp_NUMBER (4)
> Call Stack (most recent call first):
>   C:/msys64/home/sebo/zephyr/cmake/app/boilerplate.cmake:220 (include)
>   CMakeLists.txt:2 (include)
> 
> 
> CMake Error at C:/msys64/home/sebo/zephyr/cmake/version.cmake:37 (math):
>   math cannot parse the expression: "( << 24) + ( << 16) + ( << 8) + ":
>   syntax error, unexpected exp_SHIFTLEFT, expecting exp_PLUS or exp_MINUS or
>   exp_OPENPARENT or exp_NUMBER (4)
> Call Stack (most recent call first):
>   C:/msys64/home/sebo/zephyr/cmake/app/boilerplate.cmake:220 (include)
>   CMakeLists.txt:2 (include)
> 
> 
> CMake Error at C:/msys64/home/sebo/zephyr/cmake/version.cmake:39 (to_hex):
>   to_hex Function invoked with incorrect arguments for function named: to_hex
> Call Stack (most recent call first):
>   C:/msys64/home/sebo/zephyr/cmake/app/boilerplate.cmake:220 (include)
>   CMakeLists.txt:2 (include)
> 
> 
> CMake Error at C:/msys64/home/sebo/zephyr/cmake/version.cmake:40 (to_hex):
>   to_hex Function invoked with incorrect arguments for function named: to_hex
> Call Stack (most recent call first):
>   C:/msys64/home/sebo/zephyr/cmake/app/boilerplate.cmake:220 (include)
>   CMakeLists.txt:2 (include)
> 
> 
> CMake Error at C:/msys64/home/sebo/zephyr/cmake/kconfig.cmake:93 (message):
>   File not found:
>   /home/sebo/zephyr/samples/hello_world/C:/msys64/home/sebo/zephyr/boards/arm/nrf52_pca10040/nrf52_pca10040_defconfig
> Call Stack (most recent call first):
>   C:/msys64/home/sebo/zephyr/cmake/app/boilerplate.cmake:222 (include)
>   CMakeLists.txt:2 (include)
> 
> 
> -- Configuring incomplete, errors occurred!
> 
> sebo@SEBO-WIN MINGW64 ~/zephyr/samples/hello_world/build
> $
> 


I added some debug prints to inspect the path that is corrupted:

message( "${ZEPHYR_BASE}/VERSION")
file(READ ${ZEPHYR_BASE}/VERSION ver)
message( "${ZEPHYR_BASE}/VERSION")

I can't explain that "file(READ [...])" is prefixing the file with "/home/sebo/zephyr/samples/hello_world/"

--- a/doc/getting_started/installation_win.rst
+++ b/doc/getting_started/installation_win.rst
@@ -59,18 +59,17 @@ packages from their respective websites.
 
       choco feature enable -n allowGlobalConfirmation
 
-#. Install CMake and DTC:
+#. Install CMake:
 
    .. code-block:: console
 
       choco install cmake --installargs 'ADD_CMAKE_TO_PATH=System'
-      choco install dtc-msys2 --version 1.4.4
 
 #. Install the rest of the tools:
 
    .. code-block:: console
 
-      choco install git python ninja gperf
+      choco install git python ninja dtc-msys2 gperf
 
 #. Close the Command Prompt window.
 
@@ -211,6 +210,16 @@ environment for Windows. Follow the steps below to set it up:
 
       pacman -Syu
       pacman -S git cmake make gcc dtc diffutils ncurses-devel python3 gperf
+
+#. Compile :program:`Ninja` from source (Ninja is not available as
+   an MSYS2 package) and install it:
+
+   .. code-block:: console
+
+      git clone git://github.com/ninja-build/ninja.git && cd ninja
+      git checkout release
+      ./configure.py --bootstrap
+      cp ninja.exe /usr/bin/
 
 #. From within the MSYS2 MSYS Shell, clone a copy of the Zephyr source
    into your home directory using Git.  (Some Zephyr tools require

